### ISSUE TITLE
Mailbox has closed and banning all peers

### ISSUE THREAD

**Issue by ilblackdragon** [nearprotocol/nearcore#2305]
Running `nearup betanet` via Docker on my laptop via VPN (so might have network issues):

Observing, `near_network::peer_manager: Failed sending message: Mailbox has closed` error.
Post that we also banning this peer `Try to ban a disconnected peer: ed25519:ydgzeXHJ5Xyt7M1gXLxqLBW1Ejx6scNV5Nx2pxFM8su`, which is somewhat unexpected.

```
Mar 24 14:45:29.249  INFO stats: #       0 Downloading headers 0% -/4  5/2/40 peers ⬇ 2.0kiB/s ⬆ 0.5kiB/s 0.00 bps 0 gas/s CPU: 2%, Mem: 24.9 MiB
Mar 24 14:45:32.710 ERROR near_network::peer_manager: Failed sending message: Mailbox has closed
Mar 24 14:45:39.251  INFO stats: #       0 Downloading headers 0% -/4  4/1/40 peers ⬇ 1.2kiB/s ⬆ 0.4kiB/s 0.00 bps 0 gas/s CPU: 2%, Mem: 25.1 MiB
Mar 24 14:45:42.710  WARN network: Try to ban a disconnected peer: ed25519:ydgzeXHJ5Xyt7M1gXLxqLBW1Ejx6scNV5Nx2pxFM8su
Mar 24 14:45:49.255  INFO stats: #       0 Downloading headers 0% -/4  4/1/40 peers ⬇ 1.2kiB/s ⬆ 0.4kiB/s 0.00 bps 0 gas/s CPU: 2%, Mem: 25.1 MiB
Mar 24 14:45:52.719 ERROR near_network::peer_manager: Failed sending message: Mailbox has closed
Mar 24 14:45:59.257  INFO stats: #       0 Downloading headers 0% -/4  3/1/40 peers ⬇ 0.5kiB/s ⬆ 0.3kiB/s 0.00 bps 0 gas/s CPU: 2%, Mem: 25.9 MiB
Mar 24 14:46:02.728  WARN network: Try to ban a disconnected peer: ed25519:6DSjZ8mvsRZDvFqFxo8tCKePG96omXW7eVYVSySmDk8e
Mar 24 14:46:09.259  INFO stats: #       0 Downloading headers 0% -/4  3/1/40 peers ⬇ 0.5kiB/s ⬆ 0.2kiB/s 0.00 bps 0 gas/s CPU: 2%, Mem: 25.9 MiB
Mar 24 14:46:12.736 ERROR near_network::peer_manager: Failed sending message: Mailbox has closed
Mar 24 14:46:19.262  INFO stats: #       0 Downloading headers 0% -/4  2/1/40 peers ⬇ 16 B/s ⬆ 28 B/s 0.00 bps 0 gas/s CPU: 2%, Mem: 25.3 MiB
Mar 24 14:46:22.737  WARN network: Try to ban a disconnected peer: ed25519:7PGseFbWxvYVgZ89K1uTJKYoKetWs7BJtbyXDzfbAcqX
Mar 24 14:46:29.264  INFO stats: #       0 Downloading headers 0% -/4  2/1/40 peers ⬇ 16 B/s ⬆ 28 B/s 0.00 bps 0 gas/s CPU: 2%, Mem: 25.3 MiB
Mar 24 14:46:32.748 ERROR near_network::peer_manager: Failed sending message: Mailbox has closed
```

Note, my GCloud node runs fine at the same time, so it's definitely some issue with degraded network. But it reproduced very well - both inside and outside of the docker.

**Comment by mfornet** [comment]
It looks we are trying to establish connection with a peer that have been banned!

On `ban_peer` the actor is [stopped](https://github.com/nearprotocol/nearcore/blob/master/chain/network/src/peer.rs#L273), so definitely if try to establish connection with it again I expect that `Mailbox has closed` will happens. 

I think the issue is because we are not handling properly banned peers. We are lacking tests around banning mechanism so I'll try to add some and hopefully catch this issue.

**Comment by bowenwang1996** [comment]
Based on the logs it seems that the node didn't receive any headers (possibly due to connection issue) and we ban peers in this case.

**Comment by mfornet** [comment]
> Based on the logs it seems that the node didn't receive any headers (possibly due to connection issue) and we ban peers in this case.

@bowenwang1996 Why did we ban this peer exactly, for not receiving headers?

Anyway, there seems to be a bug handling banned peers, that we are probably trying to connect to them, but that peer actor is closed so it fails with such message.

**Comment by bowenwang1996** [comment]
So that a peer cannot grieve you.

**Comment by ilblackdragon** [comment]
Ok, here are more data:
 1. It takes 1m10s for me to download 512 headers. To do that I disabled banning / stalling.
 2. We ban peer on stalling duration (30s currently), it closes connection and remove actor.
 3. We send to that peers another BlockHeaderRequest message after banning it - hence "Mailbox has closed" message.
 4. We also try to ban that peer again :)

Fixes:
 - Expand stalling time to 120s (I looked at [grin code](https://github.com/mimblewimble/grin/blob/master/servers/src/grin/sync/header_sync.rs#L145) and they are using 120s as well)
 - There is a problem in header_sync_due that it will continue requesting from the peer it just banned. Fix that.
